src/fable/Fable.Core/npm/fable-core.ts

```diff
@@ -943,16 +943,22 @@ class FString {
     return FString.padLeft(str, len, ch, true);
   }
 
-  static remove(str: string, args1: number, args2?: number) {
-    var start = 0;
-    var len = 0;
-    if (args2 === undefined) {
-      len = args1;
+  static remove(str: string, startIndex: number, count?: number) {
+    if (count === undefined) {
+      if (startIndex >= str.length) {
+        throw "startIndex must be less than length of string";
+      }
+    } else {
+      if (startIndex >= str.length || count > str.length) {
+        throw "Index and count must refer to a location within the string."
+      }
+    }
+
+    if (count === undefined) {
+      return str.slice(0, startIndex);
     } else {
-      start = args1;
-      len = args2;
+      return str.slice(0, startIndex) + str.substring(startIndex + count);
```

Comment:
We can put the two operations in the same line a là JS:
```js
return str.slice(0, startIndex) + (typeof count === "number" ? str.substr(startIndex + count) : "");
```

> I usually use JS `.substr` because it has the same semantics as .NET `.Substring` (the second argument is the length not an index). But here it doesn't really matters because there's only one argument.